message.go

```diff
@@ -47,3 +48,12 @@ func (msg *Message) ShouldAck() bool {
 
 	return msg.Subscription.AckMode() != AckAuto
 }
+
+func (msg *Message) Read(p []byte) (int, error) {
+	if len(msg.Body) == 0 {
+		return 0, io.EOF
+	}
+	n := copy(p, msg.Body)
+	msg.Body = msg.Body[n:]
```

Comment:
Is there a reason for assigning the body to itself?

Reply:
Indeed, copy works with chunks, so if p has "n" bytes only "n" bytes will be copied to "p", than we need to trim body, than read it until io.EOF

Reply:
🆒